temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -20,55 +20,87 @@
 
 package io.temporal.internal.worker;
 
-import com.google.protobuf.util.Timestamps;
+import static io.temporal.internal.worker.LocalActivityResult.failed;
+
+import com.google.common.base.Preconditions;
 import com.uber.m3.tally.Scope;
 import com.uber.m3.tally.Stopwatch;
 import com.uber.m3.util.ImmutableMap;
-import io.temporal.api.common.v1.RetryPolicy;
+import io.grpc.Deadline;
+import io.temporal.api.enums.v1.RetryState;
+import io.temporal.api.enums.v1.TimeoutType;
 import io.temporal.api.failure.v1.Failure;
+import io.temporal.api.failure.v1.TimeoutFailureInfo;
 import io.temporal.api.workflowservice.v1.PollActivityTaskQueueResponse;
-import io.temporal.api.workflowservice.v1.RespondActivityTaskFailedRequest;
+import io.temporal.api.workflowservice.v1.PollActivityTaskQueueResponseOrBuilder;
 import io.temporal.common.RetryOptions;
 import io.temporal.failure.ApplicationFailure;
 import io.temporal.failure.FailureConverter;
 import io.temporal.internal.common.ProtobufTimeUtils;
+import io.temporal.internal.common.RetryOptionsUtils;
 import io.temporal.internal.logging.LoggerTag;
 import io.temporal.internal.statemachines.ExecuteLocalActivityParameters;
 import io.temporal.serviceclient.MetricsTag;
 import io.temporal.worker.MetricsType;
 import io.temporal.worker.WorkerMetricsTag;
+import io.temporal.workflow.Functions;
 import java.time.Duration;
 import java.util.Objects;
-import java.util.Optional;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.TimeUnit;
-import java.util.function.BiFunction;
+import java.util.concurrent.*;
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.slf4j.MDC;
 
 final class LocalActivityWorker implements SuspendableWorker {
   private static final Logger log = LoggerFactory.getLogger(LocalActivityWorker.class);
 
-  @Nonnull private SuspendableWorker poller = new NoopSuspendableWorker();
+  // RETRY_STATE_IN_PROGRESS shows that it's not the end
+  // for this local activity execution from the workflow point of view.
+  // It's also not conflicting with any other situations
+  // and uniquely identifies the reach of the local retries
+  // and a need to schedule a timer.
+  private static final RetryState LOCAL_RETRY_LIMIT_RETRY_STATE =
+      RetryState.RETRY_STATE_IN_PROGRESS;
```

Comment:
Why re-bind this here?

Reply:
It used in couple of places and I wanted to write a long comment on this value choice. And I don't want the comment to be duplicated.